pwndbg/auxv.py

```diff
@@ -80,11 +81,26 @@
     37: "AT_L3_CACHESHAPE",
 }
 
+
 sys.modules[__name__].__dict__.update({v: k for k, v in AT_CONSTANTS.items()})
 
 
+from typing import Protocol
+
+
+class AUXVProtocol(Protocol):
+    AT_PHDR: Optional[int]
+    AT_BASE: Optional[int]
+    AT_PLATFORM: Optional[str]
+    AT_ENTRY: Optional[int]
+    AT_RANDOM: Optional[int]
+    AT_EXECFN: Optional[str]
+    AT_SYSINFO: Optional[int]
+    AT_SYSINFO_EHDR: Optional[int]
+
+
 class AUXV(Dict[str, Union[int, str]]):
```

Comment:
Shouldnt this inherit from AUXVProtocol?

Reply:
No, Protocols use structural subtyping, meaning as long as the structure can match the protocol, it doesn't actually need to inherit from it. Implementing `__getattr__` is why this class matches the structure, as it will handle any missing attributes.

It is allowed to inherit from the Protocol, but in this case we can't do so without explicitly defining each field in `AUXV`, which takes away the benefit of using the dictionary. But it's also more clear, which is why I haven't decided if the approach in the previous commit where we just add some `@property`s for each field is a better solution.